packages/react-card/Spec.md

```diff
@@ -0,0 +1,403 @@
+| ----------- | ------------------------------------------------------------------------ |
+| Card        | Wrapper for the main content of a card.                                  |
+| CardHeader  | Optional header slot for the card.                                       |
+| CardBody    | Main slot of the Card                                                    |
+| CardPreview | Previewing the document/article presented in the card through an image   |
+| CardFooter  | Optional footer slot for the card. (non-applicable for horizontal cards) |
+
+### Card
+
+#### Anatomy
+
+![visual anatomy of the Card component](./assets/Card.png)
+
+#### API
+
+| Property    | Values                                                                               | Default    | Purpose                                                                                   |
+| ----------- | ------------------------------------------------------------------------------------ | ---------- | ----------------------------------------------------------------------------------------- |
+| orientation | `vertical`, `horizontal`                                                             | `vertical` | Orientation of the card                                                                   |
+| size        | `smallest`, `smaller`, `small`, `medium`, `large`                                    | `medium`   | Define the minimum size of the card. Smaller sizes only apply to horizontal card          |
+| scale       | `fixed`, `auto-width`, `auto-height`, `auto`, `fluid-width`, `fluid-height`, `fluid` | `auto`     | Manages how the card handles it's scaling depending on the content                        |
+| onClick     | function                                                                             | undefined  | Function to call when clicking the card. Makes the card assume the interactive appearance |
+| appearance  | `filled`, `filled-alternative`, `outline`, `subtle`                                  | `filled`   | Define the appearance of the card                                                         |
+| selectable  | boolean                                                                              | false      | Makes the card selectable by adding a checkbox to the _Actions_ area                      |
+| selected    | boolean                                                                              | false      | Set to `true` if card is selected                                                         |
+| expandable  | boolean                                                                              | false      | Allow card to expand to show whole content                                                |
+| context     | _slot_                                                                               | undefined  | Slot for the context menu, when right-clicking the card                                   |
```

Comment:
the word context might be a bit confusing - it usually suggest the React context, not the onContextMenu. Maybe we can rename this to `contextMenu`. Also, you plan to use the same slot for the ... (three dots) menu? Because that might be slightly different from the menu that appears on right click. Lastly, do we need any change in the Menu compoennt or MenuList so that it can be integrated in the slot?

Reply:
That's a good point, I will rename this to `contextMenu` as suggested.
For the three dots, there isn't enough information on the design spec that clarifies this, as only the horizontal card has it. As such, my thoughts were that that it's up to the user to add any additional menu they want and that this only covers the right-click/long press scenarios.
Regarding `Menu`/`MenuList`, I am unaware. As long as it enables us to have a menu in one place and apply the trigger wherever, all is good.